ush/python/pygfs/task/land_analysis.py

```diff
@@ -105,14 +110,34 @@ def prepare_GTS(self) -> None:
         if os.path.isfile(f"{os.path.join(localconf.DATA, output_file)}"):
             rm_p(output_file)
 
-        # execute BUFR2IODAX to convert GTS bufr data into IODA format
+        # execute BUFR2IODAX to convert adpsfc bufr data into IODA format
         yaml_file = f"bufr_adpsfc_snow.yaml"
         if not os.path.isfile(f"{os.path.join(localconf.DATA, yaml_file)}"):
             logger.exception(f"{yaml_file} not found")
             raise FileNotFoundError(f"{os.path.join(localconf.DATA, yaml_file)}")
         exe = Executable(self.task_config.BUFR2IODAX)
         exe.add_default_arg(os.path.join(localconf.DATA, f"{yaml_file}"))
 
+        logger.info(f"Executing {exe}")
+        try:
+            exe()
+        except OSError:
+            raise OSError(f"Failed to execute {exe}")
+        except Exception:
+            raise WorkflowException(f"An error occured during execution of {exe}")
+
+        output_file = f"{localconf.OPREFIX}snocvr_snow.nc4"
+        if os.path.isfile(f"{os.path.join(localconf.DATA, output_file)}"):
+            rm_p(output_file)
+
+        # execute BUFR2IODAX to convert snocvr bufr data into IODA format
+        yaml_file = f"bufr_snocvr.yaml"
+        if not os.path.isfile(f"{os.path.join(localconf.DATA, yaml_file)}"):
+            logger.exception(f"{yaml_file} not found")
+            raise FileNotFoundError(f"{os.path.join(localconf.DATA, yaml_file)}")
+        exe = Executable(self.task_config.BUFR2IODAX)
+        exe.add_default_arg(os.path.join(localconf.DATA, f"{yaml_file}"))
+
```

Comment:
I have a suggestion to remove some repetition here. Please see the patch on this file if you could kindly test.
[diff.txt](https://github.com/NOAA-EMC/global-workflow/files/12563299/diff.txt)

Reply:
I just tested this, and it worked perfect. The updates were committed to the repo. Thanks @aerorahul for your suggestions.